src/main/java/net/fabricmc/loom/MixinAnnotationProcessorExtension.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * This file is part of fabric-loom, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package net.fabricmc.loom;
+
+import java.io.File;
+import java.util.AbstractMap;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.gradle.api.Action;
+import org.gradle.api.InvalidUserDataException;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.UnknownTaskException;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.plugins.ExtraPropertiesExtension;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.util.PatternSet;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A gradle extension to configure mixin annotation processor.
+ */
+public class MixinAnnotationProcessorExtension {
```

Comment:
Any reason this is its own extension, couldnt it just be part of looms? so:

```
loom {
   mixin {
   }
}
```

Idk why someone would use mixin gradle with loom but I dont think we need to have this clash?

Reply:
There is no technical difficulty with this. However, I think making them separate will be cleaner for writing build scripts. And also touching LoomGradleExtension is a pain for me. 

About the clashing, how about I re-name it to `loom-mixin`. Or if you really want I can make it inside LGE. What's your idea?

Reply:
It should still be its own class, but you can just have a `mixin(Action<MixinAnnotationProcessorExtension>)` method in LGE.

LGE needs a good cleanup, there is a lot of stuff in there that doesnt belong there, but thats out of scope here.

Reply:
Then I think I will rename `mixin` to `loom-mixin` and add `mixin(Action<MixinAnnotationProcessorExtension>)` into LGE, so people can choose whatever they like. How do you think?

Reply:
I don't see why this is in need of a separate extension in the first place, when would it not be possible to access the `loom` extension but it would `loomMixin`?

Reply:
To be honest, I just don't know how to register an `extension` without a name in gradle. Is this even something possible? The main point here is that I don't want to put new code into LGE since it is hard to work with.

Reply:
@Chocohead any suggestion?

Reply:
My PR (https://github.com/FabricMC/fabric-loom/pull/431) to refactor LGE should help make this easier, it really shouldnt be its own extension.

Reply:
done.